### PR TITLE
Fix handling of malformed `sizes` and `heights` strings

### PR DESCRIPTION
**Pull request by chasefinch**
Some of the test output files for server-side rendering contained malformed css.

The comment above the test element said "removes invalid sizes", but in `parseSizesSpec.js` it seems that invalid sizes definition should fail in general. I can't think of reasonable logic to salvage the invalid sizes in the test input files, so I added the invalid sizes string as a test in parseSizesSpec.js, and removed it from the output files.

### CODE REVIEW

File: packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_sizes_attribute_to_css/input.html [hunk 1]
```diff
@@ -25,14 +25,5 @@
       src=https://acme.org/image1.png
       width=400>
     </amp-img>
-    <!-- removes invalid sizes -->
```
Comment (sebastianbenz) [inline packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_sizes_attribute_to_css/input.html:28]: Is this now covered by a different test case?

File: packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_heights_attribute_to_css/input.html [hunk 2]
```diff
@@ -15,13 +15,5 @@
       src=https://acme.org/image1.png
       width=320>
     </amp-img>
-    <!-- removes invalid heights -->
```
Comment (sebastianbenz) [inline packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_heights_attribute_to_css/input.html:18]: Is this case covered by a different test case?

**Review by sebastianbenz — COMMENTED** [review]
Thanks! I think it's still important to test the invalid case in the transformer tests, just to be sure that this won't break the transformer.

File: packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_sizes_attribute_to_css/input.html (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_sizes_attribute_to_css/input.html:28]: Ignore previous comment, however, I'd prefer to keep this.

File: packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_heights_attribute_to_css/input.html (hunk 2 above)
Comment (sebastianbenz) [reply packages/optimizer/spec/transformers/valid/ServerSideRendering/converts_heights_attribute_to_css/input.html:18]: ignore comment

**Comment by chasefinch** [comment]
Leaving the invalid tests in causes the expected output to preserve the boilerplate from the input; Is that okay?

**Comment by sebastianbenz** [comment]
Good point. Then we should move it to a different file.

**Review by sebastianbenz — APPROVED** [review]
Thanks a lot!

**Pull request merged**